src/Compilers/CSharp/Portable/FlowAnalysis/FlowAnalysisPass.cs

```diff
@@ -43,7 +43,7 @@ internal class FlowAnalysisPass
             if (method.ReturnsVoid || method.IsIterator || method.IsAsyncEffectivelyReturningTask(compilation))
             {
                 // we don't analyze synthesized void methods.
-                if ((method.IsImplicitlyDeclared && !method.IsScriptInitializer) ||
+                if ((method.IsImplicitlyDeclared && !method.IsScriptInitializer && (!method.IsParameterlessConstructor() || method.IsDefaultValueTypeConstructor(requireZeroInit: true))) ||
```

Comment:
>(!method.IsParameterlessConstructor() || method.IsDefaultValueTypeConstructor(requireZeroInit: true)) [](http://example.com/codeflow?start=83&length=101)

It looks like this condition will be false for a synthesized parameter-less constructor of a class. Therefore, we are going to ```Analyze``` it, which we didn't do before. I assume that was not the intent of the change.

Reply:
Good catch, thanks. I've updated the `if` condition.

I'm open to suggestions on how to improve the readability of the `if` though. It feels like the call to `IsDefaultValueTypeConstructor(bool requireZeroInit)` is the complicated part.
One possibility is to change that method to return a value that describes the default value constructor (such as the `enum` below), but that might mean performing extra work for callers that don't care about `requireZeroInit`.

```csharp
enum ValueTypeConstructorKind
{
    None,
    Explicit,
    ImplicitZeroInit,
    ImplicitWithFieldInitializers,
}
```